docs/examples/Rust/llm_dataset_creation.rst

```diff
@@ -0,0 +1,16 @@
+Creating text dataset for LLM training using Lance in Rust
+----------------------------------------------------------
+
+In this example, we will demonstrate how to achieve the Python LLM dataset creation shown in :doc:`../Python/llm_dataset_creation` in Rust.
+
+This example will show how to:
+
+1. Download and process a text dataset in parts from huggingface
+2. Tokenize the text data with a custom RecordBatchReader
+3. Save it as a Lance dataset using Lance API
```

Comment:
This example is slightly different from the python example.  In the python example the data is streamed (in random order) from hugging face.  I'm not sure if the hugging face datasets loader saves parquet files to disk or not in this process but if it does then it is an implementation detail.

In the rust example we first download a subset of the data from hugging face, in its entirety (not streaming).  We then shuffle as we read, within batches (localized shuffling).  This may be the best we can do given the rust hugging face API?

Either way, it might be good to call out the differences.